system/src/main.cpp

```diff
@@ -65,52 +93,118 @@ static volatile uint32_t TimingIWDGReload;
  */
 static volatile bool wasListeningOnButtonPress;
 /**
- * The lower 16-bits of the time when the button was first pushed.
+ * The lower 16-bits of the time when the button was first pressed.
  */
-static volatile uint16_t buttonPushed;
+static volatile uint16_t pressed_time;
 
 uint16_t system_button_pushed_duration(uint8_t button, void*)
 {
     if (button || network.listening())
         return 0;
-    return buttonPushed ? HAL_Timer_Get_Milli_Seconds()-buttonPushed : 0;
+    return pressed_time ? HAL_Timer_Get_Milli_Seconds()-pressed_time : 0;
 }
 
+#if Wiring_SetupButtonUX
+/* flag used to initiate system_handle_single_click() from main thread */
+static volatile bool SYSTEM_HANDLE_SINGLE_CLICK = false;
+
 static uint32_t prev_release_time = 0;
 static uint8_t clicks = 0;
 
+/* single click handler displays RSSI value on system LED */
+void system_handle_single_click()
+{
+    /*   signal strength (u-Blox Sara U2 and G3 modules)
+     *   0: < -105 dBm
+     *   1: < -93 dBm
+     *   2: < -81 dBm
+     *   3: < -69 dBm
+     *   4: < - 57 dBm
+     *   5: >= -57 dBm
+     */
+    if (SYSTEM_HANDLE_SINGLE_CLICK) {
+        SYSTEM_HANDLE_SINGLE_CLICK = false;
+        volatile system_tick_t startTime = SYSTEM_TICK_COUNTER;
+        RGB.control(true);
+        RGB.color(0,10,0);
+        int rssi = 0;
+        int bars = 0;
+#if Wiring_WiFi == 1
```

Comment:
I added this for the future.  Currently `Wiring_SetupButtonUX is set to 0` for all devices except the Electron.  Before I did that though I did test it on a photon and it worked.  We want to add a user API that allow users to OPT IN to enable this feature though, because many users are using the SETUP button for various things already (i.e. Short button presses up to 3 seconds, remember anything over 3 seconds activates Listening Mode).

I looked for a while for WiFi based RSSI values... every system is a bit different so we'll have to come up with values that seem appropriate for the Photon.